library/init/meta/async_tactic.lean

```diff
@@ -0,0 +1,49 @@
+prelude
+import init.meta.tactic
+import init.meta.interactive
+
+namespace tactic
+
+private meta def run_or_fail {α} (s : tactic_state) (tac : tactic α) : α :=
+match tac s with
+| (tactic_result.success a s) := a
+| (tactic_result.exception .α fmt _ s') :=
+  undefined_core $ to_string $ fmt () ++ format.line ++ to_fmt s'
+end
+
+meta def run_async {α : Type} (tac : tactic α) : tactic (task α) := do
+s ← read, return $ task.delay $ λ _,
+  match tac s with
+  | (tactic_result.success a s) := a
+  | (tactic_result.exception .α fmt _ s') :=
+    undefined_core $ to_string $ fmt () ++ format.line ++ to_fmt s'
+  end
+
+meta def prove_goal_async (tac : tactic unit) : tactic unit := do
+ctx ← local_context, revert_lst ctx,
+tgt ← target, tgt ← instantiate_mvars tgt,
+env ← get_env, tgt ← return $ env^.unfold_untrusted_macros tgt,
+when tgt^.has_meta_var (fail $ "goal contains metavariables"),
+params ← return tgt^.collect_univ_params,
+lemma_name ← mk_fresh_name,
```

Comment:
Ok, I did the less ugly version.  The tactic now creates names such as `C._aux_1` etc., see the test for an example.